src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -100,37 +111,30 @@ public ErlangModuleBuildOrders compute() {
     return true;
   }
 
-  public static void addPrepareDependenciesFailedMessage(CompileContext context) {
-    context.addMessage(CompilerMessageCategory.WARNING, "Failed to submit dependencies info to compiler. Parse transform failures may occur.", null, -1, -1);
+  public static void addPrepareDependenciesFailedMessage(@NotNull CompileContext context) {
+    context.addMessage(CompilerMessageCategory.WARNING, "Failed to submit dependencies info to compiler.", null, -1, -1);
   }
 
   @TestOnly
-  static ErlangModuleBuildOrderDescriptor getModuleBuildOrder(Module module) throws CyclicDependencyFoundException {
-    return getModuleBuildOrderInner(module);
+  @NotNull
+  static List<ErlangFileDescriptor> getBuildOrder(@NotNull Module module) throws CyclicDependencyFoundException {
+    return getTopologicallySortedFileDescriptors(module);
   }
 
   @Nullable
-  private static ErlangModuleBuildOrders getModuleBuildOrders(CompileContext context) {
-    Module[] modulesToCompile = context.getCompileScope().getAffectedModules();
-    ErlangModuleBuildOrders buildOrders = new ErlangModuleBuildOrders(modulesToCompile.length);
+  private static ErlangProjectBuildOrder getProjectBuildOrder(@NotNull CompileContext context) {
     try {
-      for (Module module : modulesToCompile) {
-        buildOrders.myModuleBuildOrderDescriptors.add(getModuleBuildOrderInner(module));
-      }
+      Module[] modulesToCompile = context.getCompileScope().getAffectedModules();
+      return new ErlangProjectBuildOrder(getTopologicallySortedFileDescriptors(modulesToCompile));
     }
     catch (CyclicDependencyFoundException e) {
-      context.addMessage(CompilerMessageCategory.ERROR, "Cyclic erlang module dependency detected. Check parse_transform usages.", null, -1, -1);
+      String message = "Cyclic erlang module dependency detected. Check files " +
+                       e.getFistNodeInCycle() + " and " + e.getLastNodeInCycle() +
+                       "or their dependencies(parse_transform, behaviour, include)";
+      LOG.error(message,e);
```

Comment:
This is not an error. It's a normal situation for us - we expect user to fix her code, and try again. When you log something as error, this error can be sent to JetBrains' exception analyzer - you certainly don't want that for this particular exception.

Also, fix your formatting.